io/plugins/eu.esdihumboldt.hale.io.gml/src/eu/esdihumboldt/hale/io/gml/reader/internal/wfs/WfsBackedGmlInstanceCollection.java

```diff
@@ -412,6 +454,24 @@ private void createNextIterator() {
 				throw new IllegalStateException(e.getMessage(), e);
 			}
 
+			// Download all the files into a folder for Bucket service
+			if (tmpdir != null) {
+				try {
+					URL url = nextUri.toURL();
+					StringBuilder fileName = new StringBuilder(tmpdir);
+					fileName.append("/").append(url.getPath().replaceAll("/", "."))
+							.append(UUID.randomUUID());
```

Comment:
I think a sequential numbering would be better than a random suffix. Then a user could identify, which was the first part, second part, etc.

Also, I think the file extension could matter, especially for the handling in the bucket-service later.

Reply:
Done, now appending `totalFeaturesProcessed` and `.gml` as extension.